enterprise/api_client/enterprise_catalog.py

```diff
@@ -30,6 +30,10 @@ class EnterpriseCatalogApiClient(JwtLmsApiClient):
     GET_CONTENT_METADATA_ENDPOINT = ENTERPRISE_CATALOG_ENDPOINT + '/{}/get_content_metadata'
     APPEND_SLASH = True
 
+    def __init__(self, user=None):
+        user = user if user else utils.get_enterprise_worker_user()
```

Comment:
I think it's safer to explicitly pass the user in all the time, so that we don't accidentally have places where we're using the permissions of the enterprise worker instead of a user's (potentially) lower permissions

Reply:
From the comment in standup, can we not use `utils.get_enterprise_worker_user()` in the management command?
I get that this is a previous pattern, but it feels less than ideal from a security perspective to me

Reply:
This is following the same pattern as the `EnrollmentApiClient` with using the "enterprise_worker" user if no `user` is supplied: https://github.com/edx/edx-enterprise/blob/1c7eba91041b2291dd14dca9b0f39b335d1ae542/enterprise/api_client/lms.py#L138

The `EcommerceApiClient` also uses the "ecommerce_worker" user when initializing the client within models.py, though it passes `request.user` when the client is initialized within a view.

From a security perspective, I think there's a distinction between a request made directly by a user and a service-to-service request. Taking the `get_course` method as an example, [that method is called from within a view](https://github.com/edx/edx-enterprise/blob/5ff8103fe8010bce4f19aeede64763aea4f7b7e0/enterprise/api/v1/views.py#L355), but the user's permissions have already been checked/validated by the `@permission_required` decorator on the `course_detail` endpoint. In this case, it's a service-to-service request that would be safe to use the "enterprise_worker" service user , since the requesting user's permissions have already been checked at that point.

Generally, my understanding is that Django models should typically not be `request`-aware as methods on a model are not always called from within a view associated with a HTTP request. If we want to keep passing in `request.user` to the initialization of `EnterpriseCatalogApiClient` within models.py, we are assuming a request object will exist, when it may not if the method isn't called from a view (e.g., from a management command).

If we begin to initialize `EnterpriseCatalogApiClient` directly within a view (i.e., not from a method on a model), that's when we would pass in the `request.user`.

Definitely feel free to correct me if my above understanding is incorrect, in which case, yes... we could probably use `utils.get_enterprise_worker_user()` within the management command.

Reply:
You're correct about the service to service request. I'm just thinking of if we need to consider the case where a user might have the correct permissions to the endpoint they're calling correctly, but not the endpoint that is called by the API client. That's probably not the case, and so everything is fine, but just wanted to explicitly call it out